cmd/osbuild-jobsite-builder/main.go

```diff
@@ -206,6 +208,42 @@ func (builder *Builder) HandlePopulate(w http.ResponseWriter, r *http.Request) e
 	if r.Method != "POST" {
 		return fmt.Errorf("Builder.HandlePopulate: unexpected request method")
 	}
+	storePath := path.Join(argBuildPath, "store")
+	err := os.Mkdir(storePath, 0755)
+	if err != nil {
+		return fmt.Errorf("Builder.HandlePopulate: failed to make store directory")
+	}
+
+	tarReader := tar.NewReader(r.Body)
+	for header, err := tarReader.Next(); err != io.EOF; header, err = tarReader.Next() {
+		if err != nil {
+			return fmt.Errorf("Builder.HandlerPopulate: failed to unpack sources")
+		}
+		switch header.Typeflag {
+		case tar.TypeDir:
+			if err := os.Mkdir(path.Join(storePath, header.Name), 0755); err != nil {
```

Comment:
Also I wonder if we should be a little bit paranoid and double check that Name is not funny, i.e. contains `..` or something. More for robustness than security (as the data comes from the trusted side)